### PR TITLE
Allow all formats for channel parameter in Session constructor.

### LINKED ISSUE #807: Passing channels to nidcpower.Session and nifgen.Session constructor is limited to string
We should allow lists, etc. just like we do for repeated capabilities.

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [X] I've added tests applicable for this pull request
### What does this Pull Request accomplish?
* Channel parameter to InitWithChannel to be any format that is also supported by repeated capability
* Fix missing channel parameter from nifgen.Session()

### List issues fixed by this Pull Request below, if any.
* Fix #807 

### What testing has been done?
* Unit
* System

### CODE REVIEW

File: build/templates/_converters.py.mako [hunk 1]
```diff
@@ -124,6 +124,22 @@ def convert_repeated_capabilities(repeated_capability, prefix=''):
     return [prefix + r for r in _convert_repeated_capabilities(repeated_capability, prefix)]
 
 
+def convert_repeated_capabilities_from_init(repeated_capability, encoding):
```
Comment (marcoskirsch) [inline build/templates/_converters.py.mako]: This function should not need to know where it's called from.
I'd rather give it a more generic name or get rid of it and join from where it's called.

`convert_repeated_capabilities_from_init`  -> `convert_repeated_capabilities_to_comma_delimited_string`
and
`convert_repeated_capabilities` -> `convert_repeated_capabilities_to_list`

File: src/nidcpower/system_tests/test_system_nidcpower.py [hunk 2]
```diff
@@ -318,3 +324,13 @@ def test_set_get_vi_int_64_attribute(session):
     session.channels['0'].active_advanced_sequence_step = 1
     read_advanced_sequence_step = session.channels['0'].active_advanced_sequence_step
     assert read_advanced_sequence_step == 1
+
+
+def test_channel_format_types():
+    with nidcpower.Session('4162', [0, 1], False, 'Simulate=1, DriverSetup=Model:4162; BoardType:PXIe') as simulated_session:
```
Comment (marcoskirsch) [inline src/nidcpower/system_tests/test_system_nidcpower.py]: For the `channels` parameter on both `nidcpower` and `nifgen` constructor:

* Allow `None`
* Default channels to `None` and not `""`
* Add a test.

File: src/nifgen/metadata/functions_addon.py [hunk 3]
```diff
@@ -139,7 +139,10 @@
     'WaitUntilDone':                                { 'parameters': { 1: { 'python_api_converter_name': 'convert_timedelta_to_milliseconds',
                                                                            'type_in_documentation': 'float in seconds or datetime.timedelta', }, }, },
     'InitializeWithChannels':                       { 'parameters': { 3: { 'python_api_converter_name': 'convert_init_with_options_dictionary', 
-                                                                           'type_in_documentation': 'dict', }, }, },
+                                                                           'type_in_documentation': 'dict', }, 
+                                                                      1: { 'name': 'channels',  # need to set the name so it doesn't look like a repeated capability to the codegenerator. Could also use 'is_repeated_capability': False
```
Comment (marcoskirsch) [inline src/nifgen/metadata/functions_addon.py]: Setting `is_repeated_capability = False` seems more correct. I can see us adding `channels` to the list of strings that we infer to be `is_repeated_capability = True`

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 @pytest.fixture(scope='function')
 def session():
-    with nifgen.Session('', False, 'Simulate=1, DriverSetup=Model:5433 (2CH);BoardType:PXIe') as simulated_session:
+    with nifgen.Session('', '0', False, 'Simulate=1, DriverSetup=Model:5433 (2CH);BoardType:PXIe') as simulated_session:
```
Comment (marcoskirsch) [inline src/nifgen/system_tests/test_system_nifgen.py]: Need to test where channel isn't passed (use default) and where None is passed.

File: src/nidcpower/system_tests/test_system_nidcpower.py (hunk 2 above)
Comment (texasaggie97) [reply src/nidcpower/system_tests/test_system_nidcpower.py]: Done

File: src/nifgen/metadata/functions_addon.py (hunk 3 above)
Comment (texasaggie97) [reply src/nifgen/metadata/functions_addon.py]: Changed

File: src/nifgen/system_tests/test_system_nifgen.py (hunk 4 above)
Comment (texasaggie97) [reply src/nifgen/system_tests/test_system_nifgen.py]: Done
